### PR TITLE
Updated environment.yml 

### LINKED ISSUE #45123: BUILD: Cannot create development environment for pandas-dev using Conda Windows
### Installation check

- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).

- [X] I have read the [creating a development environment guide](https://pandas.pydata.org/pandas-docs/stable/development/contributing_environment.html#contributing-environment)

### Platform

Windows-10-10.0.22000-SP0

### Installation Method

Built from source

### pandas Version

Updated master version

### Python Version

conda 4.11.0, python - 3.9.7

### Installation Logs

<details>
Collecting git+https://github.com/pydata/pydata-sphinx-theme.git@master (from -r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1))
  Cloning https://github.com/pydata/pydata-sphinx-theme.git (to revision master) to c:\users\prith\appdata\local\temp\pip-req-build-2p3sh5ga
  Resolved https://github.com/pydata/pydata-sphinx-theme.git to commit e4ef468e2598830aef1a1d8f65acb086cc5f8e00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting pandas-dev-flaker==0.2.0
  Using cached pandas_dev_flaker-0.2.0-py2.py3-none-any.whl (20 kB)
Collecting pytest-cython
  Using cached pytest_cython-0.1.1.post0-py2.py3-none-any.whl (6.3 kB)
Requirement already satisfied: flake8>=3.8 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pandas-dev-flaker==0.2.0->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 2)) (4.0.1)
Requirement already satisfied: docutils!=0.17.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (0.17.1)
Requirement already satisfied: sphinx in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (4.3.2)
Requirement already satisfied: beautifulsoup4 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (4.10.0)
Requirement already satisfied: pytest>=2.7.3 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (6.2.5)
Requirement already satisfied: pycodestyle<2.9.0,>=2.8.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from flake8>=3.8->pandas-dev-flaker==0.2.0->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 2)) (2.8.0)
Requirement already satisfied: pyflakes<2.5.0,>=2.4.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from flake8>=3.8->pandas-dev-flaker==0.2.0->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 2)) (2.4.0)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from flake8>=3.8->pandas-dev-flaker==0.2.0->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 2)) (0.6.1)
Requirement already satisfied: attrs>=19.2.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (21.4.0)
Requirement already satisfied: iniconfig in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (1.1.1)
Requirement already satisfied: packaging in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (21.3)
Requirement already satisfied: pluggy<2.0,>=0.12 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (1.0.0)
Requirement already satisfied: py>=1.8.2 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (1.11.0)
Requirement already satisfied: toml in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (0.10.2)
Requirement already satisfied: atomicwrites>=1.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (1.4.0)
Requirement already satisfied: colorama in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (0.4.4)
Requirement already satisfied: soupsieve>1.2 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from beautifulsoup4->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.3.1)
Requirement already satisfied: imagesize in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.3.0)
Requirement already satisfied: Jinja2>=2.3 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (3.0.3)
Requirement already satisfied: snowballstemmer>=1.1 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.2.0)
Requirement already satisfied: requests>=2.5.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.26.0)
Requirement already satisfied: babel>=1.3 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.9.1)
Requirement already satisfied: sphinxcontrib-serializinghtml>=1.1.5 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.1.5)
Requirement already satisfied: setuptools in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (60.1.1)
Requirement already satisfied: sphinxcontrib-htmlhelp>=2.0.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.0.0)
Requirement already satisfied: alabaster<0.8,>=0.7 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (0.7.12)
Requirement already satisfied: sphinxcontrib-applehelp in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.0.2)
Requirement already satisfied: sphinxcontrib-jsmath in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.0.1)
Requirement already satisfied: Pygments>=2.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.10.0)
Requirement already satisfied: sphinxcontrib-devhelp in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.0.2)
Requirement already satisfied: sphinxcontrib-qthelp in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.0.3)
Requirement already satisfied: pytz>=2015.7 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from babel>=1.3->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2021.3)
Requirement already satisfied: MarkupSafe>=2.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from Jinja2>=2.3->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.0.1)
Requirement already satisfied: idna<4,>=2.5 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from requests>=2.5.0->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (3.1)
Requirement already satisfied: charset-normalizer~=2.0.0 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from requests>=2.5.0->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2.0.9)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from requests>=2.5.0->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (2021.10.8)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from requests>=2.5.0->sphinx->pydata-sphinx-theme==0.7.1->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 1)) (1.26.7)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in c:\users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages (from packaging->pytest>=2.7.3->pytest-cython->-r C:\Users\prith\Documents\VSCode_Workspace\Repos\pandas\condaenv.fm5t30ib.requirements.txt (line 3)) (3.0.6)
Building wheels for collected packages: pydata-sphinx-theme
  Building wheel for pydata-sphinx-theme (pyproject.toml): started
  Building wheel for pydata-sphinx-theme (pyproject.toml): finished with status 'error'
Failed to build pydata-sphinx-theme

Pip subprocess error:
The system cannot find the path specified.
The system cannot find the path specified.
  Running command git clone --filter=blob:none -q https://github.com/pydata/pydata-sphinx-theme.git 'C:\Users\prith\AppData\Local\Temp\pip-req-build-2p3sh5ga'       
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\prith\anaconda3\envs\pandas-dev-test\python.exe' 'C:\Users\prith\anaconda3\envs\pandas-dev-test\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Users\prith\AppData\Local\Temp\tmp3qno97ki'
       cwd: C:\Users\prith\AppData\Local\Temp\pip-req-build-2p3sh5ga
  Complete output (16 lines):
  [stb] # nodeenv does not exist.
  [stb] # Generating new nodeenv with NodeJS 14.18.1!
  [stb] # Could not find a `node` executable.
  [stb] $ python -m nodeenv --node=14.18.1
  C:\Users\prith\AppData\Local\Temp\pip-req-build-2p3sh5ga\.nodeenv
   * Install prebuilt node (14.18.1) ..... done.
  symbolic link created for C:\Users\prith\AppData\Local\Temp\pip-req-build-2p3sh5ga\.nodeenv\Scripts\nodejs.exe <<===>> node.exe
  [stb] (nodeenv) $ node --version
  error: nodeenv-unhealthy-file-not-found

  × The `nodeenv` for this project is unhealthy.
  ╰─> None

  hint: Deleting the .nodeenv directory and trying again may work.

  Link: https://sphinx-theme-builder.rtfd.io/errors/#nodeenv-unhealthy-file-not-found
  ----------------------------------------
  ERROR: Failed building wheel for pydata-sphinx-theme
ERROR: Could not build wheels for pydata-sphinx-theme, which is required to install pyproject.toml-based projects

failed

CondaEnvException: Pip failed

</details>

I am following the instructions to build environment from https://pandas.pydata.org/docs/development/contributing_environment.html

I'm creating a build environment with 
```conda env create -f environment.yml```

line 123 in environment.yml 
```
- pip:
    - git+https://github.com/pydata/pydata-sphinx-theme.git@master
```

From what I can tell, this seems to be with the latest version of [pydata-sphinx-theme@master](https://github.com/pydata/pydata-sphinx-theme). I created an environment in conda using earlier versions of pydata-sphinx-them. It works fine till 23rd Nov 2021 commit with 
https://github.com/pydata/pydata-sphinx-theme/commit/41764f57a349b58304343bbbd7b496f939a2679c
as well as
https://github.com/pydata/pydata-sphinx-theme/commit/56d5621276ce8a2fb96c091fa294a98e19479ced
After which it seems to fail with the above mentioned error. 

The issue starts from https://github.com/pydata/pydata-sphinx-theme/commit/579d7ce695cb5c2611aa7b8ab76fbc3efc5fab26 

Can somebody else confirm this? Maybe modifying the environment.yml file to an earlier commit of pydata-sphinx-theme?

Update: I've used 
``` - git+https://github.com/pydata/pydata-sphinx-theme.git@41764f5``` to create the conda environment and built pandas-dev but cannot create environment with ``` - git+https://github.com/pydata/pydata-sphinx-theme.git@master```

```
>>> import pandas
>>> pandas.__version__
'1.4.0.dev0+1520.gaf76bd5476'
```

### PR DESCRIPTION
**Pull request by prithvijitguha**
- [X] closes #45123
- [X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [X] whatsnew entry
Changed line 123 in environment.yml from 
`git+https://github.com/pydata/pydata-sphinx-theme.git@master` to `git+https://github.com/pydata/pydata-sphinx-theme.git@41764f5`
By pre commit hook `requirements-dev.txt` line 86 changed to `git+https://github.com/pydata/pydata-sphinx-theme.git@41764f5`

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -120,6 +120,9 @@ dependencies:
   - tabulate>=0.8.3  # DataFrame.to_markdown
   - natsort  # DataFrame.sort_values
   - pip:
-    - git+https://github.com/pydata/pydata-sphinx-theme.git@master
+      #issue with building environment in conda on windowswith. Issue: https://github.com/pandas-dev/pandas/issues/45123
+      #issue with pydata-sphix-theme on windows. Issue: https://github.com/pydata/pydata-sphinx-theme/issues/523
+      #using this last stable version as workaround
+    - git+https://github.com/pydata/pydata-sphinx-theme.git@41764f5
```
Comment (jreback) [inline environment.yml:126]: can we just use a released pydata-sphinx-theme?

Comment (prithvijitguha) [reply environment.yml]: Hi @jreback, 

Yes we can. Works fine with the latest stable version of pydata-sphinx-theme. I've changed it accordingly. Thank you for the review. I've removed the previous comments from environment.yml regarding the issue trackers.

**Comment by prithvijitguha** [comment]
Update: 2 checks are failing with the latest released version `0.7.1`. I'm trying an older release version, if that works. It may be need to build from source for pydata-sphinx-theme that is causing the failure. 

If the older version fails, I'll change it to an older commit.

**Comment by prithvijitguha** [comment]
Hi @jreback, 

Using the latest release version causes failure in Doc Build and Upload / Doc Build and Upload (pull_request) pipeline and posix pipeline. I believe you have to build from source for pydata-sphinx-theme that's why installing from the repo? 

The current failure is because of timeout. Let me know what you think.

**Comment by jreback** [comment]
thanks @prithvijitguha

**Pull request merged**